### ISSUE TITLE
Add `ngAttr` like `ngClass` and `ngStyle`

### ISSUE THREAD

**Issue by wardbell** [angular/angular#13940]
**I'm submitting a ...**  (check one with "x")
```
[ ] bug report
[x] feature request
[ ] support request 
```

possibly related to #13355. might help with #9121

**Current behavior**
There is an `ngStyle` that takes an object whose key/values add a style for each key and set to the corresponding value.

There is nothing comparable for attributes.  There is `[attr.xxx]` which adds-and-sets the `xxx` attribute. Unfortunately, if you have a lot of attributes to set, you'll add a lot of `[attr]` bindings. And you could have a lot of the to set when enriching the app with ARIA.

**Expected behavior**

A comparable `ngAttr` directive would do the trick.

**Minimal reproduction of the problem with instructions**

This plunker provides a suggested implementation of `NgAttr` that borrows heavily from `NgStyle`

https://plnkr.co/edit/gcoALdBgNwqgrRtWfvr2?p=preview

**What is the motivation / use case for changing the behavior?**

Make it easier to add (and remove) a large number of ARIA attributes.

Angular feels incomplete without it as there is `[class.xxx]`/`ngClass`, `[style.xxx]`/`NgStyle`, but only `[attr.xxx]`.

**Please tell us about your environment:**
All

* **Angular version:** 2.0.X
2.0.4

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web 
all

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
all

**Comment by pkozlowski-opensource** [comment]
@wardbell mind sharing one or two examples where the new directive would be shorter / easier / better to write? Is all about the `attr.` prefix? I'm not against, just would like to see impact based on real-life examples. Thnx!

**Comment by DzmitryShylovich** [comment]
Yeah, I also don't see how it would simplify #9121 for example. I would use `@Component.host`/`@HostBinding` for impl.

**Comment by wardbell** [comment]
@DzmitryShylovich `@HostBinding` doesn't apply to elements _within a template_ such as input, button, and anchor tags; `@HostBinding`  applies to the host element of the component.  They're also considerably clunkier and less capable than `NgStyle`, `NgClass`, or the proposed `NgAttr`.

The benefits are the same as for `NgStyle` and `NgClass`: you want to set/remove a set of styles/classes in a compact way with a single property.  If they make sense, surely `NgAttr` makes sense; if `NgAttr` is superfluous, why offer `NgStyle` and `NgClass`.

@pkozlowski-opensource I gave an example in my plunker wherein I set a bunch of ARIA attributes. As with classes one may wonder when those values change?  And the answer is the same: changes in model state can change several ARIA attributes at once, including what validations and descriptions should appear. The issue #9121 speaks to this need directly.

It's actually more valuable than `NgClass` because the HTML syntax for classes is a relatively compact single string whereas attributes have to be tediously repeated within an HTML tag:
```
<div class="foo bar baz">...</div>
<input aria-label="foo" aria-required="true" aria-describedby="barDef">
```

As I mentioned above, there is a very strange lack of symmetry in supporting `class.xxx`|`ngClass`, `style.xxx`|`ngStyle`, but only `attr.xxx`.  How can that be justified? What is the argument against the effectiveness and completeness of `NgAttr`?

**Comment by pkozlowski-opensource** [comment]
@wardbell I really do hear what you are saying and I understand your seeing `ngAttr` as symmetrical to `ngClass` / `ngStyle`. But IMO it is super-important to consider potential benefits vs. "price".

I had another look at your plunker and the use-case I could extract from it was: `<button aria-label="Undo" aria-describedby="descriptionUndo">&#x238c;</button>` vs. `<button #btn4 [ngAttr]="attrObj">&#x238c; (NgAttr)</button>` 

where `attrObj` in a plunker is:

```attrObj = {
    'role': 'button',
    'aria-role': 'button',
    'aria-label': 'undo',
    'aria-describedby': 'descriptionUndo'
  }
```

Now, if we take this example we should make it real-life. By this I mean that `role` and `aria-role` equals "button" are not needed on native `<button>` elements. So we could reduce the example to:

```html
<button aria-label="Undo" aria-describedby="descriptionUndo">
```

vs. 

```html
<button #btn4 [ngAttr]="attrObj">
```html

The above doesn't make sense for single buttons only but could be shorter for elements repeated in a loop. This is the only benefit I can see right now.

Now, what are the drawbacks? For my the following ones:
* perf: we pay the price of code generated code for directives;
* confusion: it introduces "one more way of doing the same thing". Consider this example: `<button ngAttr="{disabled: exp}" [disabled]="otherExp" disabled>`. Yes, it is somehow contrived but this nicely illustrates that it opens real questions when it comes to "source of truth". In the above example it is not clear how the `disabled` state is calculated.

Now, if we come back to `ngClass` / `ngStyle` - those directives are already suffering from the mentioned problems and in all honesty I would prefer that we can live without them. Consider this example: `<button class="foo" [class]="exp" ngClass="{foo: exp"}>`. Shell we see the `foo` class on a button or not?

For me the only good argument for existence of `ngClass` / `ngStyle` is the fact that we sometimes want to dynamically calculate class _name_ (as a set of classes is from the application domain): `<button [ngClass]="exp">` (but even this could be replaced by `[class]="exp"`). If you look at popular widgets libraries (that do a lot of style / class calculation) you will notice that ngClass / ngStyle is not used at all (ng-bootstrap) or very little (material). The situation with attributes is different as we _know_ attributes names upfront (those come from a fixed list).

To sum up: I don't think that we can use the fact of `ngClass` / `ngStyle` existence to justify `ngAttr`. In my view additional confusion is not worth the benefits of this new directive and based on above arguments I would prefer that we do not introduce it. Having said this I'm going to ask @mhevery to weight here.

**Comment by pkozlowski-opensource** [comment]
Closing based on https://github.com/angular/angular/pull/13958#issuecomment-276245722

**Comment by Zec2703** [comment]
I found a soluce in Angular 4 to do the same thinks that [ngClass] in HTML, but with attributes : 

<input type="text" [attr.disabled]="variable_1 === 0 ||                                                         
                                                          variable_2 === 1 ? '' : null">

Comment => If 'variable_1' is egal at '0' OR variable_2 is egal at '1', the attribute 'disabled' is active, else if 'variable_1' is NOT egal at '0' OR 'variable-2' in NOT egal at '1', the attribute 'disabled' is NOT active
